File: app/src/main/java/com/djavid/bitcoinrate/presenter/interfaces/RateFragmentPresenter.java
package com.djavid.bitcoinrate.presenter.interfaces;

import com.djavid.bitcoinrate.core.Presenter;
import com.djavid.bitcoinrate.core.Router;
import com.djavid.bitcoinrate.model.project.ChartOption;
import com.djavid.bitcoinrate.presenter.instancestate.RateFragmentInstanceState;
import com.djavid.bitcoinrate.view.interfaces.RateFragmentView;


public interface RateFragmentPresenter extends Presenter<RateFragmentView, Router, RateFragmentInstanceState> {
    void showRate(boolean refresh);
    void showRateCryptonator(boolean refresh);
    void showRateCMC(boolean refresh);
    void refresh();
    void showChart(String crypto, String country, ChartOption chartOption, boolean refresh);
}

File: app/src/main/java/com/djavid/bitcoinrate/presenter/interfaces/RateFragmentPresenter.kt
package com.djavid.bitcoinrate.presenter.interfaces

import com.djavid.bitcoinrate.core.Presenter
import com.djavid.bitcoinrate.core.Router
import com.djavid.bitcoinrate.model.project.ChartOption
import com.djavid.bitcoinrate.presenter.instancestate.RateFragmentInstanceState
import com.djavid.bitcoinrate.view.interfaces.RateFragmentView


interface RateFragmentPresenter : Presenter<RateFragmentView, Router, RateFragmentInstanceState> {
    fun showRate(refresh: Boolean)
    fun showRateCryptonator(refresh: Boolean)
    fun showRateCMC(refresh: Boolean)
    fun refresh()
    fun showChart(crypto: String, country: String, chartOption: ChartOption, refresh: Boolean)
}
